actors/evm/src/interpreter/precompiles/fvm.rs

```diff
@@ -220,3 +220,34 @@ pub(super) fn call_actor_shared<RT: Runtime>(
 
     Ok(output)
 }
+
+/// Params:
+///
+/// | Param            | Value                     |
+/// |------------------|---------------------------|
+/// | randomness_epoch | U256 - low i64             |
+/// | entropy_length   | U256 - low u32             |
+/// | entropy          | input\[32..] (right padded)|
+///
+/// any bytes in between values are ignored
+///
+/// Returns empty array if invalid randomness type
+/// Errors if unable to fetch randomness
+pub(super) fn get_randomness<RT: Runtime>(
+    system: &mut System<RT>,
+    input: &[u8],
+    _: PrecompileContext,
+) -> PrecompileResult {
+    let mut input_params = ValueReader::new(input);
+
+    let randomness_epoch = input_params.read_value()?;
+    let entropy_length: u32 = input_params.read_value()?;
+    let entropy = input_params.read_padded(entropy_length.try_into().unwrap_or(0));
```

Comment:
Are you sure we want the unwrap or zero here? `u32::MAX` will turn into 0, I think we want to fail in that case.